onnx/defs/math/defs.cc

```diff
@@ -584,7 +606,23 @@ if attribute transA is non-zero, same for B and transB.
         "beta",
         "Scalar multiplier for input tensor C",
         AttributeProto::FLOAT,
-        1.0f);
+        1.0f)
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 2)) {
+          return;
+        }
+
+        auto transAAttr = ctx.getAttribute("transA");
+        bool transA = transAAttr ? static_cast<int>(transAAttr->i()) != 0 : false;
+        auto transBAttr = ctx.getAttribute("transB");
+        bool transB = transBAttr ? static_cast<int>(transBAttr->i()) != 0 : false;
+
+        *ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape()->add_dim() =
+          ctx.getInputType(0)->tensor_type().shape().dim(transA ? 1 : 0);
+        *ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape()->add_dim() =
+          ctx.getInputType(1)->tensor_type().shape().dim(transB ? 1 : 0);
```

Comment:
I think the problem is here. should be `transB ? 0 : 1`

According to the spec, without any transpose, we should have shapes like A = M x K, B = K x N,